src/drawers/drawer.ts

```diff
@@ -5,238 +5,130 @@
+ * A Drawer is a stateful class that holds one SVGDrawer and one CanvasDrawer, and can switch between
+ * the two.
  */
 export class Drawer {
-  private _renderArea: SimpleSelection<void>;
   private _canvas: d3.Selection<HTMLCanvasElement, any, any, any>;
+  private _canvasDrawer: CanvasDrawer;
 
-  protected _svgElementName: string;
-  protected _className: string;
-  protected _dataset: Dataset;
+  private _dataset: Dataset;
 
-  private _cachedSelectionValid = false;
-  private _cachedSelection: SimpleSelection<any>;
-  private _cachedSelectionNodes: d3.BaseType[];
+  private _svgRootG: d3.Selection<SVGElement, any, any, any>;
+  private _svgDrawer: SVGDrawer;
 
   /**
    * A Drawer draws svg elements based on the input Dataset.
    *
    * @constructor
    * @param {Dataset} dataset The dataset associated with this Drawer
    */
-  constructor(dataset: Dataset) {
+  constructor(dataset: Dataset, svgDrawer: SVGDrawer, canvasDrawer: CanvasDrawer) {
+    this._canvasDrawer = canvasDrawer;
     this._dataset = dataset;
-    this._svgElementName = "path";
+    this._svgDrawer = svgDrawer;
   }
 
-  public canvas(): d3.Selection<HTMLCanvasElement, any, any, any>;
   /**
-   * Sets the renderArea() to null and tells this Drawer to draw to the specified canvas instead.
-   * @param canvas
+   * Use SVG rendering from now on. This will append a new <g> element to the renderArea and draw
+   * DOM elements into it.
    */
-  public canvas(canvas: d3.Selection<HTMLCanvasElement, any, any, any>): this;
-  public canvas(canvas?: d3.Selection<HTMLCanvasElement, any, any, any>): any {
-    if (canvas === undefined) {
-      return this._canvas;
-    }
-    canvas = coerceExternalD3(canvas);
-    this._canvas = canvas;
-    this._renderArea = null;
-    this._cachedSelectionValid = false;
-    return this;
-  }
-
-  /**
-   * Retrieves the renderArea selection for the Drawer.
-   */
-  public renderArea(): SimpleSelection<void>;
-  /**
-   * Sets the canvas() to null and tells this Drawer to draw to the specified SVG area instead.
-   *
-   * @param {d3.Selection} Selection containing the <g> to render to.
-   * @returns {Drawer} The calling Drawer.
-   */
-  public renderArea(area: SimpleSelection<void>): this;
-  public renderArea(area?: SimpleSelection<void>): any {
-    if (area === undefined) {
-      return this._renderArea;
-    }
-    area = coerceExternalD3(area);
-    this._renderArea = area;
+  public useSVG(parentRenderArea: SimpleSelection<any>) {
     this._canvas = null;
-    this._cachedSelectionValid = false;
-    return this;
+    this._svgRootG = parentRenderArea.append<SVGGElement>("g");
   }
 
   /**
-   * Removes the Drawer and its renderArea
+   * Use Canvas rendering from now on. This will remove the <g> element created by useSVG, if it exists.
    */
-  public remove() {
-    if (this.renderArea() != null) {
-      this.renderArea().remove();
+  public useCanvas(canvas: d3.Selection<HTMLCanvasElement, any, any, any>) {
```

Comment:
given this mutual exclusion, maybe the HTMLElement (svg or canvas) should be managed by the respective drawer object.

for example, below, there is the .selection() method but that only applies to svg drawers that have a svg comeponent.